frontend/src/views/components/LanguageSwitcher/LanguageSwitcher.component.tsx

```diff
@@ -27,14 +28,22 @@ type Language = {
 
 function LanguageSwitcher() {
   const languages: Language[] = useLanguages();
-  const { setParam } = useRouterParams();
-  const changeLang = (code: (typeof AVAILABLE_LANGUAGES)[number]) => {
-    setParam('lang', code);
+  const router = useRouter();
+
+  const setLanguage = (locale) => {
+    const { pathname, asPath, query } = router;
+    // change just the locale and maintain all other route information including href's query
+    router.push({ pathname, query }, asPath, { locale });
+  };
```

Comment:
Can you set the cookie as well so that the user's language preference is kept in the future?

See an example here: https://github.com/Vizzuality/heco-invest/blob/develop/frontend/containers/layouts/language-selector/component.tsx#L27-L39